src/main/java/org/betonquest/betonquest/quest/condition/height/HeightCondition.java

```diff
@@ -0,0 +1,59 @@
+package org.betonquest.betonquest.quest.condition.height;
+
+import org.betonquest.betonquest.api.profiles.OnlineProfile;
+import org.betonquest.betonquest.api.quest.condition.online.OnlineCondition;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.instruction.variable.VariableNumber;
+import org.betonquest.betonquest.instruction.variable.location.VariableLocation;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Condition to check if a player is at a certain height or lower.
+ */
+public class HeightCondition implements OnlineCondition {
+
+    /**
+     * The height to check for.
+     */
+    @Nullable
+    private final VariableNumber height;
+
+    /**
+     * The height in a location to check.
+     */
+    @Nullable
+    private final VariableLocation location;
```

Comment:
This was not what i was thinking of, it could actually be that this is an anti pattern, i have to think a bit about it